script.py

```diff
@@ -0,0 +1,270 @@
+"""To use this script, install the provided requirements.txt file and
+supply a secrets.py file with the required parameters listed below.
+
+JSON records will be written into the specified file and HTML reports
+will be written into an outputs filter under the provided directory.
+"""
+
+
+from datetime import datetime, timedelta
+from jinja2 import Template
+from ua_parser import user_agent_parser
+
+import collections
+import json
+import os
+import requests
+import secrets
+import smtplib
+
+
+# This is the absolute path to the directory where you'd like
+# the JSON and HTML files to be written.
+# The template HTML file should also be placed in this
+# directory.
+DIR = secrets.DIR if hasattr(secrets, "DIR") else os.getcwd()
+# This is the name of the JSON file that you'd like to write to
+FILENAME = secrets.FILENAME
+# This is the username for your LA1 login
+USERNAME = secrets.USERNAME
+# This is the password for your LA1 login
+PASSWORD = secrets.PASSWORD
+# Email address from which you'd like to send viewer information
+FROM_EMAIL = secrets.FROM_EMAIL if hasattr(secrets, "FROM_EMAIL") else None
+# Readable name for who is sending the report (e.g. Production Team)
+FROM_NAME = secrets.FROM_NAME
+# Email address to which you'd like to send viewer information
+TO_EMAIL = secrets.TO_EMAIL if hasattr(secrets, "TO_EMAIL") else None
+# Prefix to the subject in email reports
+EMAIL_SUBJECT_PREFIX = secrets.EMAIL_SUBJECT_PREFIX
+
+if not USERNAME or not PASSWORD:
+    raise Exception("Username and password to LA1 required.")
+print(
+    f"""Directory in use: {DIR}
+Filename in use: {FILENAME}
+"""
+)
+
+if not FROM_EMAIL or not TO_EMAIL:
+    print("Skipping email since from or to emails are not provided.")
+else:
+    print("Email Template")
+    print(
+        f"""From: {FROM_NAME} <{FROM_EMAIL}>
+To: {TO_EMAIL}
+Subject: {EMAIL_SUBJECT_PREFIX} [EVENT NAME WILL GO HERE]
+"""
+    )
+
+
+def send_email(event):
+    # This server is Gmail's Restricted SMTP Server and will only work for
+    # sending emails to G Suite or Gmail accounts.
+    # See https://support.google.com/a/answer/176600?hl=en
+    server = smtplib.SMTP("aspmx.l.google.com", 25)
+    server.sendmail(
+        FROM_EMAIL,
+        TO_EMAIL,
+        f"""From: {FROM_NAME} <{FROM_EMAIL}>
+To: {TO_EMAIL}
+Subject: {EMAIL_SUBJECT_PREFIX}
```

Comment:
I did like the idea of still keeping the event name in the subject. Since event names tend to be static, that would let it thread on different types of events folks may have rather than dumping them all into the same thread. Would also allow forwarding/filtering on event name. I only wanted to take out the highly unique timestamp part.